mfs/dir.go

```diff
@@ -361,29 +346,24 @@ func (d *Directory) AddChild(name string, nd ipld.Node) error {
 		return err
 	}
 
-	err = d.unixfsDir.AddChild(d.ctx, name, nd)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return d.unixfsDir.AddChild(d.ctx, name, nd)
 }
 
-func (d *Directory) sync() error {
+func (d *Directory) syncCache(clean bool) error {
```

Comment:
Is there any situation where we would not want `clean = true`? Seems like we should always clear the cache.

Reply:
And there's also the issue that this still allows the cache to grow forever until sync. And such sync may take minutes or hours...